librz/core/disasm.c

```diff
@@ -5211,6 +5211,9 @@ RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, i
 	rz_cons_break_push (NULL, NULL);
 	for (i = idx = ret = 0; addrbytes * idx < len && ds->lines < ds->l; idx += inc, i++, ds->index += inc, ds->lines++) {
 		ds->at = ds->addr + idx;
+		if (ds->pdf && ds->at >= rz_analysis_function_max_addr (ds->pdf)) {
+			break;
```

Comment:
I think the right place to fix this would be here:
https://github.com/rizinorg/rizin/blob/da4703d757f25b51c520deb83deb55e9ee98af45/librz/core/cmd_print.c#L5389-L5403
You can see it is starting to disassemble at `fcn->addr` which is already wrong because the addr of a function is its entrypoint, not necessarily its lowest address. For how `pdf` is meant to work, it should probably just disassemble from the min address to the max address and the whole logic there about `RZ_MAX (linearsz, realsz);` and so on is probably superfluous.

Reply:
I think users would be surprised though if `pdf` starts from min address rather than entrypoint, for functions where min address != entrypoint.

Reply:
This would still have been the correct place to check, even if starting to disassemble at the entrypoint. Imho the merged fix is wrong, or at least hiding the real issue.